zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobUpdateProcessor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.job;
+
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedRejectionWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedResponseWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.state.immutable.JobState;
+import io.camunda.zeebe.engine.state.immutable.ProcessingState;
+import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.scheduler.clock.ActorClock;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+
+public class JobUpdateProcessor implements TypedRecordProcessor<JobRecord> {
+
+  private static final String NO_JOB_FOUND_MESSAGE =
+      "Expected to update job with key '%d', but no such job was found";
+  private static final String NEGATIVE_RETRIES_MESSAGE =
+      "Expected to update job with key '%d' with a positive amount of retries, "
+          + "but the amount given was '%d'";
+
+  private final JobState jobState;
+  private final StateWriter stateWriter;
+  private final TypedRejectionWriter rejectionWriter;
+  private final TypedResponseWriter responseWriter;
+
+  public JobUpdateProcessor(final ProcessingState state, final Writers writers) {
+    jobState = state.getJobState();
+    stateWriter = writers.state();
+    rejectionWriter = writers.rejection();
+    responseWriter = writers.response();
+  }
+
+  @Override
+  public void processRecord(final TypedRecord<JobRecord> command) {
+    final long key = command.getKey();
+    final int retries = command.getValue().getRetries();
+    final long timeout = command.getValue().getTimeout();
+    final var job = jobState.getJob(key, command.getAuthorizations());
+
+    if (job == null) {
+      rejectionWriter.appendRejection(
+          command, RejectionType.NOT_FOUND, NO_JOB_FOUND_MESSAGE.formatted(key));
+      responseWriter.writeRejectionOnCommand(
+          command, RejectionType.NOT_FOUND, NO_JOB_FOUND_MESSAGE.formatted(key));
+      return;
+    }
+
+    // Handle retries
+    if (retries > 0) {
+      job.setRetries(retries);
+      stateWriter.appendFollowUpEvent(key, JobIntent.RETRIES_UPDATED, job);
+      responseWriter.writeEventOnCommand(key, JobIntent.RETRIES_UPDATED, job, command);
```

Comment:
I'm not sure what do you mean ? Using 2 intents is something that I choose to do for reusing the same applier of the 2 other command, so the behaviour (at state level) is the same

Reply:
👍 Yes, we are writing two follow-up events using `stateWriter::appendFollowUpEvent`, reusing the RETRIES_UPDATED and TIMEOUT_UPDATED intents.
💭 The thing I am wondering about is the `responseWriter::writeEventOnCommand` that we use twice. We are writing two responses back if retries and timeout are updated. Does that make sense? We receive one command UPDATE and reply with two responses. I am not too familiar with how the responses are digested, to be honest. So just wondering if this could become an issue.

Reply:
Hey @tmetzke I had a quick chat with Nico about this, please have a look at my commit message 👍